src/game/fleet/combat/round.rs

```diff
@@ -100,15 +99,19 @@ fn attack (battle: &mut Battle, fid: FactionID, attacker: &FleetSquadron, round_
     })
 }
 
-fn pick_target_squadron(battle: &Battle, faction_id: FactionID, attacker: &FleetSquadron) -> Result<(FleetSquadron, f64)> {
-    let (opponent_squadrons, attack_coeff) = || -> Result<(Vec<FleetSquadron>, f64)> {
+fn pick_target_squadron(battle: &Battle, faction_id: FactionID, attacker: &FleetSquadron) -> Result<(FactionID, FleetSquadron, f64)> {
```

Comment:
J'ai proposé une PR #46  qui modifie le code de cette fonction. Dedans je modifie pas mal tout le code de la grosse boucle `for`. En gros, j'enlève les `collect()` et les `clone()` inutiles qui font perdre en perf, et je permet à une escadrille de "ne pas attaquer" si elle ne trouve aucun target attaquable. On peut refuser ma PR à cause du changement de paradigme, mais si on la refuse pour ça, il faudrait quand même garder les modifs liées aux perfs.